src/mame/machine/xavix.cpp

```diff
@@ -487,6 +488,63 @@ WRITE8_MEMBER(xavix_state::dispctrl_posirq_y_w)
 
 /* Per Game IO port callbacks */
 
+
+CUSTOM_INPUT_MEMBER(xavix_i2c_state::i2c_r)
+{
+	return m_i2cmem->read_sda();
+}
+
+CUSTOM_INPUT_MEMBER(xavix_i2c_cart_state::i2c_r)
+{
+	return m_i2cmem->read_sda();
+}
+
+CUSTOM_INPUT_MEMBER(xavix_i2c_lotr_state::camera_r) // seems to be some kind of camera status bits
+{
+	return machine().rand();
+}
+
+CUSTOM_INPUT_MEMBER(xavix_i2c_bowl_state::camera_r) // seems to be some kind of camera status bits
+{
+	return machine().rand();
+}
+
+CUSTOM_INPUT_MEMBER(xavix_ekara_state::ekara_multi0_r)
+{
+	switch (m_extraioselect & 0x7f)
+	{
+	case 0x01: return (m_extra0->read() & 0x01) >> 0; break;
+	case 0x02: return (m_extra0->read() & 0x04) >> 2; break;
+	case 0x04: return (m_extra0->read() & 0x10) >> 4; break;
+	case 0x08: return (m_extra0->read() & 0x40) >> 6; break;
+	case 0x10: return (m_extra1->read() & 0x01) >> 0; break;
+	case 0x20: return (m_extra1->read() & 0x04) >> 2; break;
+	case 0x40: return (m_extra1->read() & 0x10) >> 4; break;
```

Comment:
This looks a lot like it's a wired-AND configuration - why not bitswap/invert/or/test?  That probably better matches how the hardware behaves.